tests/typecheck/test_basics.yml

```diff
@@ -0,0 +1,13 @@
+- case: created_instance_is_of_correct_type
+  main: |
+    from django.db import models
+    import factory.django
+
+    class Book(models.Model):
+        ...
+
+    class BookFactory(factory.django.DjangoModelFactory[Book]):
+        ...
+
+    instance = BookFactory.create()
+    reveal_type(instance)  # N: Revealed type is "main.Book"
```

Comment:
Nice one 🙌🏻 

Question: does it also work for `BookFactory()`?

Reply:
No, it doesn't.

In simple cases we just need to add types for `__new__`, but in this case with metaclass i haven't figured out yet how to make it work.

Reply:
@youtux have you managed to get this working in you types?

Reply:
no, it seems to be a mypy bug: https://github.com/python/mypy/issues/14122

Reply:
Here is a workaround until the mypy issue is solved:

```python
T = TypeVar("T")


class BaseFactory(Generic[T]):
    """Factory base support for sequences, attributes and stubs."""

    # Backwards compatibility
    UnknownStrategy = errors.UnknownStrategy
    UnsupportedStrategy = errors.UnsupportedStrategy

    if TYPE_CHECKING:
        def __new__(cls, *args, **kwargs) -> T:
            ...
    else:
        def __new__(cls, *args, **kwargs):
            """Would be called if trying to instantiate the class."""
            raise errors.FactoryError('You cannot instantiate BaseFactory')
```

`mypy` will currently not support this, but will most probably be supported soon (see https://github.com/python/mypy/pull/16020). `pyright` (and thus Pylance) supports it.

Reply:
Updated my code here, it should work now with pyright and when that PR merged - it will work with mypy too.